apollo-adapters/src/jvmMain/kotlin/com/apollographql/apollo3/adapter/JavaTimeAdapters.kt

```diff
@@ -28,33 +28,33 @@ object JavaInstantAdapter : Adapter<Instant> {
 }
 
 /**
- * An [Adapter] that converts an ISO 8601 String like "2010-06-01T22:19:44.475Z" to/from
+ * An [Adapter] that converts an ISO 8601 String like "2010-06-01" to/from
  * a [java.time.LocalDate]
  *
  * It requires Android Gradle plugin 4.0 or newer and [core library desugaring](https://developer.android.com/studio/write/java8-support#library-desugaring).
  */
 object JavaLocalDateAdapter : Adapter<LocalDate> {
   override fun fromJson(reader: JsonReader, customScalarAdapters: CustomScalarAdapters): LocalDate {
-    return formatter.parse(reader.nextString()!!, LocalDate::from)
+    return LocalDate.parse(reader.nextString()!!)
   }
 
   override fun toJson(writer: JsonWriter, customScalarAdapters: CustomScalarAdapters, value: LocalDate) {
-    writer.value(formatter.format(value))
+    writer.value(value.toString())
```

Comment:
I think it should be in the same format as `KotlinxLocalDateAdapter`.

https://github.com/apollographql/apollo-kotlin/blob/e243c1fc83f16857ac79eabd0794a0ab11bcc596/apollo-adapters/src/commonMain/kotlin/com/apollographql/apollo3/adapter/KotlinxTimeAdapters.kt#L43-L57